packages/server/tests/acceptance/rpc_batch1.spec.ts

```diff
@@ -1544,21 +1547,26 @@ describe('@api-batch-1 RPC Server Acceptance Tests', function () {
           await Assertions.assertPredefinedRpcError(error, sendRawTransaction, true, relay, [signedTx, requestDetails]);
         });
 
-        it('@release fail "eth_getTransactionReceipt" on precheck with wrong nonce error when sending a tx with a higher nonce', async function () {
-          const nonce = await relay.getAccountNonce(accounts[2].address, requestId);
-
-          const transaction = {
-            ...default155TransactionData,
-            to: parentContractAddress,
-            nonce: nonce + 100,
-            gasPrice: await relay.gasPrice(requestId),
-          };
-
-          const signedTx = await accounts[2].wallet.signTransaction(transaction);
-          const error = predefined.NONCE_TOO_HIGH(nonce + 100, nonce);
-
-          await Assertions.assertPredefinedRpcError(error, sendRawTransaction, true, relay, [signedTx, requestDetails]);
-        });
+        if (!useAsyncTxProcessing) {
+          it('@release fail "eth_getTransactionReceipt" on precheck with wrong nonce error when sending a tx with a higher nonce', async function () {
+            const nonce = await relay.getAccountNonce(accounts[2].address, requestId);
+
+            const transaction = {
+              ...default155TransactionData,
+              to: parentContractAddress,
+              nonce: nonce + 100,
+              gasPrice: await relay.gasPrice(requestId),
+            };
+
+            const signedTx = await accounts[2].wallet.signTransaction(transaction);
+            const error = predefined.NONCE_TOO_HIGH(nonce + 100, nonce);
+
+            await Assertions.assertPredefinedRpcError(error, sendRawTransaction, true, relay, [
+              signedTx,
+              requestDetails,
+            ]);
+          });
+        }
```

Comment:
Note for reviewers: currently disable this test of fast return is turned on.

Open for discussion: With fast return enabled, no errors are thrown—only precheck errors. Currently, prechecks do not account for higher nonce transactions.

Should we remove the test case for higher nonce transactions, or should we retain the existing logic and assert that the transaction receipt retrieved using the returned transaction hash will be null?

Reply:
Just random thoughts.

In the Ethereum ecosystem, there is no such thing as nonce too high. We can submit a transaction with whatever nonce above the current one we want, and the transaction will reach the mempool. Once it's in the mempool, the validator will pick the transaction with a nonce of current + 1, so if there are enough transactions filling the gap between, the transaction will be executed in the near future. They always will be executed ordered by nonce.

On another hand, in the Hedera ecosystem, there is no such thing as mempool. That means if we send a transaction, it's being sent straight to the node for its execution. It turned out that if we asynchronously send N transactions without timeout between them, there is a high chance of failures due to missing "native" nonce ordering like in Ethereum. If we off-chain "ordered" them with some timeout between them (e.g. 50ms), they might succeed if there is no unexpected race condition.

Also, we should think about submitting multiple off-chain ordered transactions within a single block. In that case, if there is a nonce too high check precheck, we will limit the user to submit only one transaction per block because "the expected" nonce is fetched from the mirror node where it's up-to-date with the latest block, not the current one.

IMO nonce too high precheck must not be added, and throwing nonce too high as an error might be misleading for the end-user when multiple transactions are submitted and can be removed (or changed to WRONG_NONCE which is the actual error from the services). Also, I'm not sure the `NONCE_TOO_HIGH` error even exists in the Ethereum JSON-RPC context.

@Nana-EC @quiet-node What do you think? Does it make sense to you to rename or remove the current nonce too high error?

Reply:
Hey @natanasow, thanks for the comment! That makes complete sense. In fact, on the Relay, we agreed not to reject requests with higher nonces anymore and removed the `NONCE_TOO_HIGH` precheck. This test is believed to be a leftover from that work. 

Currently, the test still passes and throws a `NONCE_TOO_HIGH` error, but that happens because the transaction is rejected at the Services level, not on the Relay. Since there’s no longer a high nonce precheck on the Relay, the transaction gets forwarded to Services, where it is ultimately rejected with a `WRONG_NONCE` status code. The Relay then interprets this error, compares the current account nonce, and throws a `NONCE_TOO_HIGH` error when the transaction nonce is higher than the account nonce. As a result, the test passes with the expected error but with an incorrect underlying intention.

Now the point you've raised is indeed a solid and exellent point showing a gap in our EVM equivalence goals. However, since the Relay no longer enforces the `NONCE_TOO_HIGH` check and the rejection occurs at the Services level, it might be more appropriate to move this conversation to Services to bring broader attention to this gap.

Reply:
@Nana-EC Should I keep this test (which is only enabled when fast-return is off) for additional coverage, or should I remove it to clarify that the Relay no longer fails at higher nonce values?